compute/zstevx2.c

```diff
@@ -0,0 +1,584 @@
+    workStack.offd = offd;
+    workStack.jobtype = jobtype;
+    workStack.range = range;
+    workStack.il = il;
+    workStack.iu = iu;
+    workStack.stein_arrays = stein_arrays;
+
+    /* Find actual min and max eigenvalues. */
+    Bound_MinMax_Eigvalue(workStack.diag, workStack.offd, workStack.N, &globMinEval, &globMaxEval);
+    if (0) fprintf(stderr, "%s:%i globMinEval=%.15f, globMaxEval=%.15f, vl=%.15f vu=%.15f\n",
+           __func__, __LINE__, globMinEval, globMaxEval, vl, vu);
+
+    int evLessThanVL=0, evLessThanVU=n, nEigVals=0;
+    if (range == PlasmaRangeV) {
+        /* We don't call Sturm if we already know the answer. */
+        if (vl >= globMinEval) evLessThanVL=plasma_zlaneg2(diag, offd, n, vl);
+        else vl = globMinEval; /* optimize for computing step size. */
+
+        if (vu <= globMaxEval) evLessThanVU=plasma_zlaneg2(diag, offd, n, vu);
+        else vu = nexttoward(globMaxEval, DBL_MAX);  /* optimize for computing step size */
+        /* Compute the number of eigenvalues in [vl, vu). */
+        nEigVals = (evLessThanVU - evLessThanVL);
+        if (0) fprintf(stderr, "%s:%i evLessThanVU=%i, evLessThanVL=%i, nEigVals=%i.\n", 
+               __func__, __LINE__, evLessThanVU, evLessThanVL, nEigVals);
+
+         workStack.baseIdx = evLessThanVL;
+    } else {
+        /* PlasmaRangeI: iu, il already vetted by code above. */
+        nEigVals = iu+1-il; /* The range is inclusive. */
+        /* We still bisect by values to discover eigenvalues, though. */
+        vl = globMinEval;
+        vu = nexttoward(globMaxEval, DBL_MAX); /* be sure to include globMaxVal. */
+        workStack.baseIdx = 0; /* There are zero eigenvalues less than vl. */
+    }
+
+    /* if we just need to find the count of eigenvalues in a value range, */
+    if (jobtype == PlasmaCount) {
+        pFound[0] = nEigVals;
+        return(PlasmaSuccess);
+    }
+
+    /* Now if user's K (arg 4) isn't enough room, we have a problem. */
+    if (k < nEigVals) {
+        return -4;             /* problem with user's K value. */
+    }   
+
+    /* We are going into discovery. Make sure we have arrays. */
+    if (pVal == NULL) return -12;   /* pointers cannot be null. */
+    if (pMul == NULL) return -13;
+    if (jobtype == PlasmaVec && pVec == NULL) return -14;   /* If to be used, cannot be NULL. */
+
+    /* handle value range. */
+    /* Set up workStack controls. */
+    workStack.eigenvalues = nEigVals;
+    workStack.finished = 0;
+    workStack.pVal = pVal;
+    workStack.pMul = pMul;
+    workStack.pVec = pVec;
+    
+    /* Create a bracket per processor to kick off the work stack.
+     * here, low and high are the range values.
+     * We need low+max_threads*step = hi-step. ==>
+     * (hi-low) = step*(max_threads+1) ==>
+     * step = (hi-low)/(max_threads+1).
+     */
+
+    plasma_complex64_t step = (vu - vl)/(max_threads+1);
+    plasma_complex64_t prevUpper = vl;
+    for (i=0; i<max_threads; i++) {
```

Comment:
There should be line call to `calloc()`. In fact, it should be `malloc()`. The threads just add an offset based on `omp_get_thread_num()`